yang/frr-staticd.yang

```diff
@@ -0,0 +1,150 @@
+
+  import ietf-routing-types {
+    prefix "rt-types";
+  }
+
+  import ietf-routing {
+    prefix "rt";
+  }
+
+  import ietf-ipv4-unicast-routing {
+    prefix "v4ur";
+  }
+
+  import ietf-ipv6-unicast-routing {
+    prefix "v6ur";
+  }
+
+  import frr-vrf {
+    prefix frr-vrf;
+  }
+
+  organization
+    "Free Range Routing";
+
+  contact
+    "FRR Users List: <mailto:frog@lists.frrouting.org>
+     FRR Development List: <mailto:dev@lists.frrouting.org>";
+
+  description
+    "This module defines a model for managing FRR staticd information.
+     This YANG module augments the ietf-routing with additional
+     nexthop information";
+
+  revision 2019-12-03 {
+    description
+      "Initial revision.";
+  }
+
+  grouping nexthop-common-attributes {
+    leaf table-id {
+      type uint32 {
+        range "1..4294967295";
+      }
+      description
+        "Routing table id to retrieve.";
+    }
+    leaf distance {
+      type uint8  {
+        range "1..255";
+      }
+      description
+        "Admin distance based on routing protocol.";
+    }
+    leaf mpls-label {
+      type rt-types:mpls-label;
+      description
+        "Nexthop's MPLS label stack.";
+    }
+    leaf nexthop-vrf {
+      type string {
+        length "1..36";
+      }
+      description
+        "The nexthop vrf name, if different from the route.";
+    }
+  }
+
+  grouping frr-staticd-nexthop {
+
+    choice next-hop-type {
+      case  gateway {
+        leaf gw-interface {
+          type string {
+            length "1..16";
+          }
+          description
+            "Name of the outgoing interface.";
+        }
+        leaf is-onlink {
+          type empty;
+            description
+              "Nexthop is directly connected.";
+        }
+      }
+      case ifname {
+        leaf outgoing-interface {
+          type string {
+            length "1..16";
+          }
+          description
+            "Name of the outgoing interface.";
+        }
+      }
+      case Null0 {
+        leaf Null0 {
+          type boolean;
+	  default false;
+            description
+              "Null interface";
+        }
+      }
+      case reject {
+        leaf reject {
+          type boolean;
+	  default false;
+            description
+              "Emit an ICMP unreachable when matched";
+        }
+      }
+      case blackhole {
+        leaf blackhole {
+          type boolean;
+	  default false;
+            description
+              "Silently discard pkts when matched";
+        }
+      }
+    }
+
+    uses nexthop-common-attributes;
+
+  }
+
+
+  augment "/frr-vrf:lib/frr-vrf:vrf"
+    + "/rt:routing/rt:control-plane-protocols/"
+    + "rt:control-plane-protocol/rt:static-routes/v4ur:ipv4/"
+    + "v4ur:route/v4ur:next-hop/v4ur:next-hop-options/"
+    + "v4ur:next-hop-list/v4ur:next-hop-list/v4ur:next-hop"
+  {
+    description
+      "Augment static route configuration 'next-hop-list'.";
+    uses frr-staticd-nexthop;
```

Comment:
I feel like this is the wrong way to go about it. We should have a list of nexthops modeled in one place and then other places that need to reference a nexthop can use a leafref. Thoughts?

Reply:
@qlyoung I think we need to go over call and discuss this? We wanted to keep it as close to IETF as possible keeping extensions in mind if we have to use leafref then we may not be able to use IETF defined model. Though many things in IETF defined yang model are not being used today and we will be using deviate and delete statements. I think if we can go over call and converge faster on the same.

Reply:
Sounds good